sound/soc/sof/intel/hda-ipc.c

```diff
@@ -254,12 +255,13 @@ int hda_dsp_ipc_get_window_offset(struct snd_sof_dev *sdev, u32 id)
 }
 
 void hda_ipc_msg_data(struct snd_sof_dev *sdev,
-		      struct snd_pcm_substream *substream,
+		      struct snd_sof_pcm_stream *sps,
 		      void *p, size_t sz)
 {
-	if (!substream || !sdev->stream_box.size) {
+	if (!sps || !sdev->stream_box.size) {
 		sof_mailbox_read(sdev, sdev->dsp_box.offset, p, sz);
 	} else {
+		struct snd_pcm_substream *substream = sps->substream;
```

Comment:
dont we need to check for substream != NULL after this here?

Reply:
@ranj063 here's an explanation for this: https://github.com/thesofproject/linux/pull/2703/files#r618389757

Reply:
@ranj063 I completely removed this patch from the PR.